src/mlpack/methods/ann/loss_functions/multilabel_softmargin_loss.hpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file multilabel_softmargin_loss.hpp
+ * @author Anjishnu Mukherjee
+ *
+ * Definition of the Multi Label Soft Margin Loss function.
+ * It is a criterion that optimizes a multi-label one-versus-all loss based on
+ * max-entropy, between input x and target y of size (N, C) where N is the
+ * batch size and C is the number of classes.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_ANN_LOSS_FUNCTION_MULTILABEL_SOFTMARGIN_LOSS_HPP
+#define MLPACK_ANN_LOSS_FUNCTION_MULTILABEL_SOFTMARGIN_LOSS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class MultiLabelSoftMarginLoss
+{
+ public:
+  /**
+   * Create the MultiLabelSoftMarginLoss object.
+   *
+   * @param weight A manual rescaling weight given to each class. Weight is a
+   *               (1, C) vector where C is the number of classes.
+   * @param num_classes The number of classes, C.
+   * @param reduction Specifies the reduction to apply to the output. If false,
+   *                  'mean' reduction is used, where sum of the output will be
+   *                  divided by the number of elements in the output. If
+   *                  true, 'sum' reduction is used and the output will be
+   *                  summed. It is set to true by default.
+   */
+  MultiLabelSoftMarginLoss(arma::mat weight,
+                           const size_t num_classes,
```

Comment:
We use camel-case in mlpack for declarations. Kindly refer to our style [guidelines](https://github.com/mlpack/mlpack/wiki/DesignGuidelines).

Reply:
Aaah! Force of habit. 
Thanks for noticing, changed it!